pkgsMusl.glibc does not compile on master due to warning

## Issue description

Recently (I suspect with the GCC 9 upgrade that may introduce new warnings) `pkgsMusl.glibc` broke on master with error:

```
./../include/libc-symbols.h:471:26: error: ‘__EI___errno_location’ specifies less restrictive attributes than its target ‘__errno_location’: ‘const’, ‘nothrow’ [-Werror=missing-attributes]
  471 |   extern __typeof (name) __EI_##name \
      |                          ^~~~~
./../include/libc-symbols.h:475:29: note: in expansion of macro ‘__hidden_ver1’
  475 | #  define hidden_def(name)  __hidden_ver1(__GI_##name, name, name);
      |                             ^~~~~~~~~~~~~
./../include/libc-symbols.h:539:32: note: in expansion of macro ‘hidden_def’
  539 | # define libc_hidden_def(name) hidden_def (name)
      |                                ^~~~~~~~~~
errno-loc.c:28:1: note: in expansion of macro ‘libc_hidden_def’
   28 | libc_hidden_def (__errno_location)
      | ^~~~~~~~~~~~~~~
errno-loc.c:24:1: note: ‘__EI___errno_location’ target declared here
   24 | __errno_location (void)
      | ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

### Steps to reproduce

`nix-build -A pkgsMusl.glibc`

## Technical details

nixpkgs commit 85a9743f135b3687281efc5881baaab8ca2a8e09

CC @dtzWill

--- a/pkgs/development/libraries/glibc/default.nix
+++ b/pkgs/development/libraries/glibc/default.nix
@@ -54,9 +54,10 @@ callPackage ./common.nix { inherit stdenv; } {
         # Fix -Werror build failure when building glibc with musl with GCC >= 8, see:
         # https://github.com/NixOS/nixpkgs/pull/68244#issuecomment-544307798
         (stdenv.lib.optional stdenv.hostPlatform.isMusl "-Wno-error=attribute-alias")
-        (stdenv.lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
+        (stdenv.lib.optionals ((stdenv.hostPlatform != stdenv.buildPlatform) || stdenv.hostPlatform.isMusl) [
           # Ignore "error: '__EI___errno_location' specifies less restrictive attributes than its target '__errno_location'"
           # New warning as of GCC 9
+          # Same for musl: https://github.com/NixOS/nixpkgs/issues/78805
           "-Wno-error=missing-attributes"
         ])
       ]);

